packages/lesswrong/components/layout/PersistentHomepage.tsx

```diff
@@ -0,0 +1,79 @@
+'use client';
+
+import React, { ReactNode, useEffect, useState, createContext, useContext, Activity as ActivityExport } from 'react';
+import { useLocation } from '@/lib/routeUtil';
+import LWHome from '../common/LWHome';
+
+// Context to mark when the UltraFeed has been viewed (scrolled into view)
+const UltraFeedViewedContext = createContext<(() => void) | null>(null);
+
+export const useMarkUltraFeedAsViewed = () => {
+  return useContext(UltraFeedViewedContext);
+};
+
+// Activity might not be available at runtime even though types declare it
+// Try multiple fallbacks: stable Activity -> unstable_Activity -> Symbol fallback
+// TODO: Fix all this, issues because Activity not being properly available
+const Activity = (
+  ActivityExport ?? 
+  (React as any).unstable_Activity ?? 
+  (Symbol.for('react.activity') as any)
+) as React.ComponentType<{ children?: ReactNode; mode?: 'visible' | 'hidden' }>;
```

Comment:
idk what's happening here but Activity should be available as of 19.2 so I suspect this is either unnecessary, or that something else is going wrong if it is